test/src/main.rs

```diff
@@ -20,6 +20,13 @@ use std::sync::atomic::Ordering;
 use std::sync::Arc;
 use std::time::{Duration, Instant};
 
+struct TestResult {
+    spec_name: String,
+    status_name: String,
```

Comment:
a minor suggestion: create a enum for TestResult status,
```
enum TestResultStatus {
  Pass,
  Panic,
  Fail
}
```

and implement `Ord` for this enum, then we can remove the `status_weight` from `TestResult` and sort the results by this status field.

Reply:
Done, please take another look, thx